docs/using-react-redux/static-types.md

```diff
@@ -0,0 +1,106 @@
+---
+id: static-types
+title: Type safety and React-Redux
+hide_title: true
+sidebar_label: Type safety and React-Redux
+---
+
+# Type safety and React-Redux
+
+The [`react-redux` type definitions](https://npm.im/@types/react-redux) export some helpers to make it easier to write typesafe interfaces between your redux store and your React components.
+
+## Typing the useSelector hook
+
+If you manually type your selector functions, `useSelector` will return the same type as your selector
+
+```ts
+// selectIsOn always returns a boolean, so isOn is typed as a boolean
+const isOn = useSelector(selectIsOn)
+```
+
+Otherwise `useSelector` requires you to manually type the state argument every time you use it.
+
+```ts
+const isOn = useSelector((state: MyStateType) => state.isOn)
+```
+
+If you'd like, you can define a typed `useSelect` hook using a helper type
+
+```ts
+// reducer.ts
+import { useSelector, TypedUseSelectorHook } from 'react-redux'
+
+interface MyStateType {
+  isOn: boolean
+}
+
+export const useTypedSelector: TypedUseSelectorHook<MyStateType> = useSelector
+
+// my-component.tsx
+import { useTypedSelector } from './reducer.ts'
+
+const isOn = useSelector(state => state.isOn)
+```
+
+## Typing the `connect` higher order component
+
+React-redux exposes a helper type, `ConnectedProps`, that can extract the return type of mapStateToProps and mapDispatchToProps.
+
+```ts
+import { connect, ConnectedProps } from 'react-redux'
+
+interface MyStateType {
+  isOn: boolean
+}
+
+const connector = connect(
+  (state: MyStateType) => ({
+    isOn: state.isOn
+  }),
+  (dispatch: any) => ({
+    toggleOn: () => dispatch({ type: 'TOGGLE_IS_ON' })
+  })
+)
+
+type PropsFromRedux = ConnectedProps<typeof connector>
+```
+
+The return type of `ConnectedProps` can then be used to type your props object.
+
+```tsx
+interface Props extends PropsFromRedux {
```

Comment:
Y'know, I keep forgetting you can extend interfaces.

Reply:
Would you rather I say `type Props = PropsFromRedux & {`? I've a slight preference for types, but the codebase I work on mostly uses interfaces. I've no idea which would be easier to understand

Reply:
That's the syntax I've been using, but not sure if it matters that much.